src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -165,13 +170,41 @@ private Task StartAsyncCore()
                         return;
                     }
 
-                    _transport = _transportFactory.CreateTransport(GetAvailableServerTransports(negotiationResponse));
-                    connectUrl = CreateConnectUrl(Url, negotiationResponse);
+                    // This should only need to happen once
+                    _serverTransports = GetAvailableServerTransports(negotiationResponse);
+                    connectUrl = CreateConnectUrl(Url, negotiationResponse.ConnectionId);
                 }
 
-                _logger.StartingTransport(_transport, connectUrl);
-                await StartTransport(connectUrl);
+                foreach (var transport in AllTransports)
+                {
+                    var createNewConnectionId = false;
+                    try
+                    {
+                        if ((transport & _serverTransports & _requestedTransportType) != 0)
+                        {
+                            if (createNewConnectionId)
```

Comment:
Two things: the `createNewConnectionId` value is defined inside the loop, so this won't really work. Also, you can simplify this by just doing the "re-negotiation" at the end of the loop, in the catch block.

Reply:
Smh

> Also, you can simplify this by just doing the "re-negotiation" at the end of the loop, in the catch block.

I wanted to add the logic with flag so that we can avoid re-negotiating after the last transport has failed

Reply:
Hrm, yeah... I'm not sure I like either way of dealing with that... At the very least, fix the variable for now ;)